lib/rails/generators/rspec/templates/component_spec.rb.tt

```diff
@@ -2,6 +2,7 @@
 
 require "rails_helper"
 
+<% module_namespacing do -%>
```

Comment:
I'd prefer it here if the module namespacing was on the described class if we can do that – ideally the `RSpec.describe` block should be top-level. It's sometimes easier to wrap a spec file in a module to save having to mention it all the time, but I don't think we should default to that.

Reply:
Probably we can try:

```diff
diff --git a/lib/rails/generators/rspec/templates/component_spec.rb.tt b/lib/rails/generators/rspec/templates/component_spec.rb.tt
index 9992f946..7eff719f 100644
--- a/lib/rails/generators/rspec/templates/component_spec.rb.tt
+++ b/lib/rails/generators/rspec/templates/component_spec.rb.tt
@@ -2,8 +2,7 @@

 require "rails_helper"

-<% module_namespacing do -%>
-RSpec.describe <%= class_name %>Component, type: :component do
+RSpec.describe <%= namespaced? ? "#{namespace.name}::" : '' %><%= class_name %>Component, type: :component do
   pending "add some examples to (or delete) #{__FILE__}"

   # it "renders something useful" do
@@ -14,4 +13,3 @@ RSpec.describe <%= class_name %>Component, type: :component do
   #   )
   # end
 end
-<% end -%>
```

Reply:
Updated for test unit and rspec generator

Reply:
Awesome, thank you! 🎉